### DISCUSSION TITLE
How do I set a default value for input field?

### DISCUSSION THREAD

**Discussion by dinuone** [livewire/livewire#3717] [1. Help]
**when try to add default values to my input and I select the image which need to upload into the database input values are hidden when submitting anyone can give me suggestions**

**this is my modal**
  <form wire:submit.prevent="save" enctype="multipart/form-data">
          
              @foreach ($students as $student)
                <div class="form-group">
                  <label for="">Full Name : </label>
                  <label> <strong>{{ $student->FullName }}</strong></label>
                </div>
                <div class="form-group">
                  <input type="text"  class="form-control"  value="{{ $student->id }}"wire:model.defer="st_id">
                </div>
                <div class="form-group">
                  <label for="">Student ID : </label>
                  <label> <strong>{{ $student->student_id }}</strong></label>
                  <input type="text"  class="form-control"  value="{{ $student->student_id }}"wire:model.defer="student_id" readonly>
                  <span class="text-danger">@error('student_id')  {{ $message }} @enderror</span>
                </div>
              @endforeach
              <hr>

**Comment by ryatkins** [comment]
@dinuone - you would set your initial value on your public property and get rid of the value parameter on the input field:

`public $st_id = 5;`

` <input type="text"  class="form-control"  wire:model.defer="st_id">`

**Reply by dinuone** [reply]
I tried but its not working. still when i try to upload image then all inputs default values are hidden in the fields 

```

              @foreach ($students as $student)

              @php
                 $st_id = $student->id;
              @endphp

                <div class="form-group">
                  <label for="">Full Name : </label>
                  <label> <strong>{{ $student->FullName }}</strong></label>
                </div>
                <div class="form-group">
                  <input type="text"  class="form-control"  value="{{ $st_id }}"wire:model.defer="st_id" >
                  
                </div>
                <div class="form-group">
                  <label for="">Student ID : </label>
                  <label> <strong>{{ $student->student_id }}</strong></label>
                  <input type="text"  class="form-control"  value="{{ $student->student_id }}"wire:model.defer="student_id">
                  <span class="text-danger">@error('student_id')  {{ $message }} @enderror</span>
                </div>
              @endforeach
              <hr>
```

**Reply by ridwaanzan** [reply]
i wonder why you use value parameter in your input field?
it should be done using wire:model="your_variable"  or wire:model.defer="your_variable"

try to set the default value in your livewire controller component not in your livewire view component.

i think it will work, coz i usually using it like that.

**Reply by devcircus** [reply]
I've read through th8s thread several times but still not sure I understand. Thus is a very common situation. Set the default values on a public property in your component, then wire:model the property in the view. You can use arrays or models as well and bind to nested data.
